tests/test_rpc.c

```diff
@@ -96,32 +167,212 @@ test_lock(void **state)
+    nc_send_rpc(st->nc_sess, rpc, 1000, &msgid);
+
+    /* recieve reply, should fail since wrong permissions */
+    msgtype = nc_recv_reply(st->nc_sess, rpc, msgid, 2000, &envp, &op);
+    assert_int_equal(msgtype, NC_MSG_REPLY);
+    assert_null(op);
+    assert_string_equal(LYD_NAME(lyd_child(envp)), "rpc-error");
+
+    nc_rpc_free(rpc);
+    lyd_free_tree(envp);
+    /* TODO: Check error message, would depend on current user */
+    /* TODO: NACM tests */
+}
+
+static void
+test_rpc_commit(void **state) {
+    struct np_test *st = *state;
+    struct nc_rpc *rpc;
+    NC_MSG_TYPE msgtype;
+    uint64_t msgid;
+    struct lyd_node *envp, *op;
+
+    /* Try to close a session */
+    rpc = nc_rpc_commit(0, 0, NULL, NULL, NC_PARAMTYPE_CONST);
+    nc_send_rpc(st->nc_sess, rpc, 1000, &msgid);
+
+    /* recieve reply, should fail since wrong permissions */
     msgtype = nc_recv_reply(st->nc_sess, rpc, msgid, 2000, &envp, &op);
     assert_int_equal(msgtype, NC_MSG_REPLY);
     assert_null(op);
     assert_string_equal(LYD_NAME(lyd_child(envp)), "ok");
+
+    nc_rpc_free(rpc);
     lyd_free_tree(envp);
+    /* TODO: test funcionality */
+}
+
+static void
+test_rpc_discard(void **state) {
+    struct np_test *st = *state;
+    struct nc_rpc *rpc;
+    NC_MSG_TYPE msgtype;
+    uint64_t msgid;
+    struct lyd_node *envp, *op;
+
+    /* Try to close a session */
+    rpc = nc_rpc_discard();
+    nc_send_rpc(st->nc_sess, rpc, 1000, &msgid);
+
+    /* recieve reply, should fail since wrong permissions */
+    msgtype = nc_recv_reply(st->nc_sess, rpc, msgid, 2000, &envp, &op);
+    assert_int_equal(msgtype, NC_MSG_REPLY);
+    assert_null(op);
+    assert_string_equal(LYD_NAME(lyd_child(envp)), "ok");
 
     nc_rpc_free(rpc);
+    lyd_free_tree(envp);
+    /* TODO: test funcionality */
+}
+
+static void
+test_rpc_cancel(void **state) {
+    struct np_test *st = *state;
+    struct nc_rpc *rpc;
+    NC_MSG_TYPE msgtype;
+    uint64_t msgid;
+    struct lyd_node *envp, *op;
+
+    /* TODO: Segfaults  */
+    /* Try to close a session */
+    rpc = nc_rpc_cancel(NULL, NC_PARAMTYPE_CONST);
+    nc_send_rpc(st->nc_sess, rpc, 1000, &msgid);
 
-    nc_session_free(nc_sess2, NULL);
+    /* recieve reply, should fail since wrong permissions */
+    msgtype = nc_recv_reply(st->nc_sess, rpc, msgid, 2000, &envp, &op);
+    assert_int_equal(msgtype, NC_MSG_REPLY);
+    assert_null(op);
+    assert_string_equal(LYD_NAME(lyd_child(envp)), "rpc-error");
+
+    nc_rpc_free(rpc);
+    /* TODO: test funcionality */
 }
 
 int
 main(void)
 {
     const struct CMUnitTest tests[] = {
-        cmocka_unit_test(test_lock),
+        cmocka_unit_test(test_nc_rpc_types),
+        cmocka_unit_test(test_rpc_lock),
+        cmocka_unit_test(test_rpc_unlock),
+        cmocka_unit_test(test_rpc_get),
+        cmocka_unit_test(test_rpc_kill),
+        cmocka_unit_test(test_rpc_commit),
+        cmocka_unit_test(test_rpc_discard),
+        cmocka_unit_test(test_rpc_cancel),
```

Comment:
No need to call every test `*_rpc_*`, the whole test file is called `test_rpc` so it is implied.